### PR TITLE
feat: support drag handle in default page

### PR DESCRIPTION
**Pull request by Himself65**

### CODE REVIEW

File: packages/blocks/src/page-block/default/selection-manager.ts [hunk 1]
```diff
@@ -420,6 +423,60 @@ export class DefaultSelectionManager {
...
+              ) as BaseBlockModel;
+              const idx = currentParentModel.children.findIndex(
+                m => m.id === currentModel.id
+              );
+              const nextIdx = nextParentModel.children.findIndex(
+                m => m.id === nextModel.id
+              );
+              if (
+                recursiveFindParent(this.page, currentModel, nextModel) ||
+                recursiveFindParent(this.page, nextModel, currentModel)
```
Comment (colelawrence) [inline packages/blocks/src/page-block/default/selection-manager.ts:453]: Break this out into a named value like `const isParent___ = ` and check with `if (isParent___) return;`

(I don't know what this is checking, so obviously `isParent___` is just a stand in.

File: packages/blocks/src/page-block/default/selection-manager.ts [hunk 2]
```diff
@@ -420,6 +423,60 @@ export class DefaultSelectionManager {
...
+              ) as BaseBlockModel;
+              const nextParentModel = this.page.getParent(
+                nextModel
+              ) as BaseBlockModel;
+              const idx = currentParentModel.children.findIndex(
+                m => m.id === currentModel.id
+              );
+              const nextIdx = nextParentModel.children.findIndex(
+                m => m.id === nextModel.id
+              );
```
Comment (colelawrence) [inline packages/blocks/src/page-block/default/selection-manager.ts]: Looks like all these lines can be moved after the `if (
                recursiveFindParent(this.page, currentModel, nextModel) ||
                recursiveFindParent(this.page, nextModel, currentModel)` check

File: packages/store/src/workspace/page.ts [hunk 3]
```diff
@@ -276,6 +276,28 @@ export class Page extends Space<PageData> {
     this.updateBlock(model, props);
   }
 
+  moveBlock(
+    model: BaseBlockModel,
+    oldParent: BaseBlockModel,
+    oldIndex: number,
+    newParent: BaseBlockModel,
+    newIndex: number
+  ) {
```
Comment (colelawrence) [inline packages/store/src/workspace/page.ts]: Lots of positional arguments of the same types is usually a code smell. I'll defer to @doodlewind and other code owners for whether this is actually aligned with the codebase's maintainability guidelines.

Alternatively, pass in an
```
  opts: {
    target: BaseBlockModel,
    from: { parent: BaseBlockModel, idx: number },
    to: { parent: BaseBlockModel, idx: number }
  }
```

**Review by colelawrence — COMMENTED** [review]
A few comments on different TypeScript patterns. I do not know all the parts of this codebase, so feel free to discard any of my ideas.

**Review by doodlewind — APPROVED** [review]
Awesome! we can leave future improvements in following PRs 🍗

**Pull request merged**